plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestingSqlServer.java

```diff
@@ -119,14 +120,19 @@ private void setUpDatabase()
         execute("CREATE DATABASE " + databaseName);
 
         // Enable snapshot isolation by default to reduce flakiness on CI
-        Failsafe.with(QUERY_RETRY_POLICY)
-                .run(() -> execute(format("ALTER DATABASE %s SET ALLOW_SNAPSHOT_ISOLATION ON", databaseName)));
-        Failsafe.with(QUERY_RETRY_POLICY)
-                .run(() -> execute(format("ALTER DATABASE %s SET READ_COMMITTED_SNAPSHOT ON", databaseName)));
+        executeDatabaseSetUpWithRetry(databaseName -> format("ALTER DATABASE %s SET ALLOW_SNAPSHOT_ISOLATION ON", databaseName));
+        executeDatabaseSetUpWithRetry(databaseName -> format("ALTER DATABASE %s SET READ_COMMITTED_SNAPSHOT ON", databaseName));
 
         container.withUrlParam("database", this.databaseName);
     }
 
+    public void executeDatabaseSetUpWithRetry(Function<String, String> setupQuery)
```

Comment:
I don't think that using function here is beneficial. In use case above you have `databaseName` already defined. In other places you can call `getDatabaseName()` if you like.

Reply:
it emphasizes the purpose of the API. it's not meant to be used for anything else than database setup.

Reply:
I see, but I don't understand why is that important.

How about plain `executeWithRetry`?

Reply:
because i don't plan to introduce "execute anything with any retries" API. 
if i did, i wouldn't know when i can retry and how i can evolve the implementation. 

also, in the future, the method may need to disappear, if we retry whole database setup instead